File: problems/912/solution.py
class Solution:
    def sortArray(self, nums: List[int]) -> List[int]:
        self.mergeSort(nums)
    
        return nums
    
    def mergeSort(self, nums):
        if len(nums) > 1:
            mid = len(nums) // 2
            l = nums[:mid]
            r = nums[mid:]
            
            self.mergeSort(l)
            self.mergeSort(r)
            
            i, j, k = 0, 0, 0
            
            while i < len(l) and j < len(r):
                if l[i] < r[j]:
                    nums[k] = l[i]
                    i += 1
                else:
                    nums[k] = r[j]
                    j += 1

                k += 1
        
            while i < len(l):
                nums[k] = l[i]
                k += 1
                i += 1

            while j < len(r):
                nums[k] = r[j]
                k += 1
                j += 1

File: problems/912/solution.cpp
class Solution {
public:
    vector<int> sortArray(vector<int>& nums) {
        mergeSort(nums);
        
        return nums;
    }
    
    void mergeSort(vector<int>& nums) {
        if (nums.size() <= 1) {
            return;
        }
        
        vector<int> l, r;
        
        for (int i = 0; i < nums.size(); i++) {
            if (i < nums.size() / 2) {
                l.push_back(nums[i]);
            } else {
                r.push_back(nums[i]);
            }
        }
        
        mergeSort(l);
        mergeSort(r);
        
        int i = 0, j = 0, k = 0;
        
        while (i < l.size() && j < r.size()) {
            if (l[i] < r[j]) {
                nums[k++] = l[i++];
            } else {
                nums[k++] = r[j++];
            }
        }
        
        while (i < l.size()) {
            nums[k++] = l[i++];
        }
        
        while (j < r.size()) {
            nums[k++] = r[j++];
        }
    }
};